node-backend/app/controllers/v1/recognitionController.js

```diff
@@ -0,0 +1,346 @@
+const yup = require("yup");
+const moment = require("moment"); //eslint-disable-line node/no-extraneous-require
+const qs = require("qs"); //eslint-disable-line node/no-extraneous-require
+
+const db = require("../../models/sequelize");
+const jwtValidate = require("../../jwtTokenValidation/jwtValidation");
+const utility = require("../../utils/utility");
+const Recognitions = db.recognitions;
+const CoreValues = db.coreValues;
+const Users = db.users;
+
+const schema = yup.object().shape({
+  core_value_id: yup
+    .number({ core_value_id: "should be number" })
+    .required({ core_value_id: "required" })
+    .typeError({ core_value_id: "should be number" }),
+  text: yup.string({ text: "should be text" }).required({ text: "required" }),
+  given_for: yup
+    .number()
+    .required({ given_for: "required" })
+    .typeError({ given_for: "should be number" }),
+  given_by: yup.number().typeError({ given_by: "should be number" }),
+  given_at: yup.number().typeError({ given_at: "should be number" }),
+});
+
+const validateCoreValue = async (req, res) => {
+  const tokenData = await jwtValidate.getData(req.headers["authorization"]);
```

Comment:
token data should be first fetched and then pass on to validating functions. So that in each validation functions you don't need to decode token data again and again